### ISSUE TITLE
Add 2nd RepositoryInterface

### ISSUE THREAD

**Issue by JoeBengalen** [JoeBengalen/Cache#3]
Make 2 repository interfaces: one with only the singular function and a second (extending the first) which also included the multi functions. Using the multi function only has its advantages when getting cache data takes longer and the backend is able to fetch or store multiple entries at once.

The Pool should then check of which interface the repository derrives and act upon it.

An example would be a database repository. Longer fetches and can handle multiple cache items at once.

**Comment by JoeBengalen** [comment]
Idea: Instead of doing some logic in the pool the determine if the xxAll methods are available, wrap the Repo in a wrapper that maps the xxxAll methods to their corresponding singular methods.

``` php
class RepositoryWrapper implements RepositoryAllInterface
{
    public function __construct(RepositoryInterface $repository)
    {
        $this->repository = $repository;
    }

    public function getAll($keys)
    {
        foreach ($keys as $key) {
            $this->repository->get($key);
        }
    }
}
```

Decide whether the wrapping has to be done by the user or by the pool!

**Comment by JoeBengalen** [comment]
Also maybe the normal interface should be kept the same and the other called `SimpleRepositoryInterface` without the xxAll() methods

**Comment by JoeBengalen** [comment]
Added the second suggestion in branch https://github.com/JoeBengalen/Cache/tree/split-repo-iface-wrapper

**Comment by JoeBengalen** [comment]
I prefer the adapter over the extra logic in the Pool. Therefore I will merge the latest branch back to develop.

I am still not too happy with the name `SimpleRepositoryInterface`, but it will have to do for now.

**Comment by JoeBengalen** [comment]
solved by #5
